tools/toolchains/arm.py

```diff
@@ -416,27 +417,21 @@ def __init__(self, target, *args, **kwargs):
             self.flags['common'].append("-mfpu=fpv5-sp-d16")
             self.flags['common'].append("-mfloat-abi=hard")
 
-        if ((target.core.startswith("Cortex-M23") or
-             target.core.startswith("Cortex-M33")) and
-            not target.core.endswith("-NS")):
-            self.flags['cxx'].append("-mcmse")
-            self.flags['c'].append("-mcmse")
-
-        # Create Secure library
-        if ((target.core.startswith("Cortex-M23") or
-             target.core.startswith("Cortex-M33")) and
-            not target.core.endswith("-NS") and
-            kwargs.get('build_dir', False)):
-            build_dir = kwargs['build_dir']
-            secure_file = join(build_dir, "cmse_lib.o")
-            self.flags["ld"] += ["--import_cmse_lib_out=%s" % secure_file]
-
-        # Add linking time preprocessor macro DOMAIN_NS
-        if ((target.core.startswith("Cortex-M23") or
-             target.core.startswith("Cortex-M33")) and
-            target.core.endswith("-NS")):
-            define_string = self.make_ld_define("DOMAIN_NS", "0x1")
-            self.flags["ld"].append(define_string)
+        if CORE_ARCH[target.core] == 8:
+            # Add linking time preprocessor macro DOMAIN_NS
+            if target.core.endswith("-NS"):
+                define_string = self.make_ld_define("DOMAIN_NS", "0x1")
+                self.flags["ld"].append(define_string)
+            else:
```

Comment:
Incrementally better than current version, but as per #9460, this still isn't the final solution. Ultimately it shouldn't need to test the architecture at all, and could just be looking at "-S" or "-NS" suffixes. Then the non TrustZone version has no suffix.

Reply:
‘-S’ does not exists at present, and I don’t know the impacts of changing core names at present, online compiler will be impacted for sure. Left this decision on tools team for now